File: frontend/src/enums.ts
export enum HttpStatus {
  OK = 200,
  CREATED = 201,
  NO_CONTENT = 204,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
  FORBIDDEN = 403,
  NOT_FOUND = 404,
  LOGIN_TIMEOUT = 440,
  INTERNAL_SERVER_ERROR = 500,
}

export enum SortDirection {
  Ascending = "asc",
  Descending = "desc",
}

export enum GCSdimension {
  Latitude = "latitude",
  Longitude = "longitude",
}

export enum MeterSortByField {
  SerialNumber = "serial_number",
  RANumber = "ra_number",
  LandOwner = "land_owner_name",
  TRSS = "trss",
}

export enum MeterStatusNames {
  Installed = "Installed",
  Warehouse = "Warehouse",
  Scrapped = "Scrapped",
  Returned = "Returned",
  Sold = "Sold",
  Unknown = "Unknown",
}

export enum WellSortByField {
  Name = "name",
  RANumber = "ra_number",
  OSEPod = "osepod",
  UseType = "use_type",
  Location = "location",
}

export enum MeterHistoryType {
  Activity = "Activity",
  Observation = "Observation",
}

export enum ActivityType {
  Install = "Install",
  Uninstall = "Uninstall",
  GeneralMaintenance = "General Maintenance",
  PreventativeMaintenance = "Preventative Maintenance",
  Repair = "Repair",
  RateMeter = "Rate Meter",
  Sell = "Sell",
  Scrap = "Scrap",
  ChangeWaterUsers = "Change Water Users",
}

export enum WorkingOnArrivalValue {
  NotChecked = "not-checked",
  NotWorking = "not-working",
  Working = "working",
}

export enum WorkOrderStatus {
  Open = "Open",
  Closed = "Closed",
  Review = "Review",
}

File: api/enums.py
from enum import Enum
from api.security import scoped_user


class MeterSortByField(Enum):
    SerialNumber = "serial_number"
    RANumber = "ra_number"
    WaterUsers = "water_users"
    TRSS = "trss"


class WellSortByField(Enum):
    Name = "name"
    RANumber = "ra_number"
    OSETag = "osetag"
    UseType = "use_type"
    Location = "location"

class MeterStatus(Enum):
    #Status can be: Installed, Warehouse, Scrapped, Returned, Sold, or Unknown
    Installed = "Installed"
    Warehouse = "Warehouse"
    Scrapped = "Scrapped"
    Returned = "Returned"
    Sold = "Sold"
    Unknown = "Unknown"
    OnHold = "On Hold"

class SortDirection(Enum):
    Ascending = "asc"
    Descending = "desc"


class ScopedUser(Enum):
    Read = scoped_user(["read"])
    Admin = scoped_user(["admin"])
    OSE = scoped_user(["ose"])
    ActivityWrite = scoped_user(["activities:write"])
    WellMeasurementWrite = scoped_user(["well_measurement:write"])
    MeterWrite = scoped_user(["meters:write"])
    WellWrite = scoped_user(["well:write"])

class WorkOrderStatus(Enum):
    Open = "Open"
    Closed = "Closed"
    Review = "Review"
